climpyrical/data.py

```diff
@@ -81,29 +83,63 @@ def read_data(data_path: str, required_keys: list = None) -> xr.Dataset:
         )
 
     if required_keys is None:
-        required_keys = ["rlat", "rlon"]
+        required_keys = ["rlat", "rlon", "lat", "lon"]
 
     with xr.open_dataset(data_path) as ds:
         all_keys = list(set(ds.variables).union(set(ds.dims)))
+
         check_valid_keys(all_keys, required_keys)
         check_valid_data(ds)
 
-        if "time" in ds.keys() and ds["time"].size <= 1:
-            ds = ds.squeeze("time").drop_vars("time")
-        if "time_bnds" in ds.keys():
-            ds = ds.drop_vars("time_bnds")
-        if "rotated_pole" in ds.keys():
-            ds = ds.drop_vars("rotated_pole")
+        # keys with size > 2 are good, otherwise
+        # are superfluous (time, time_bnds, rotated_pole, etc)
+        good_keys = []
+        bad_keys = []
+        for key in all_keys:
+            if ds[key].size > 2:
+                good_keys.append(key)
+            else:
+                bad_keys.append(key)
```

Comment:
Is there a reason we even build the `good_keys` list when it doesn't get used in the method? We could save some space here by using list comprehension to just build the `bad_keys` list. On top of that I don't think `bad_keys` is the best name for them.